LEDs/StrutLayoutFactory.cs

```diff
@@ -48,6 +49,43 @@ public class StrutLayoutFactory {
     };
     private static readonly EdgeDictionary edgeDictionary;
 
+    // JKMD: these are all the points of the dome, mapped into 2d space
+    // JKMD: these are normalized to -1 to 1 coordinates in InitPoints
+    // What has become of my life
+    private static int[,] handDrawnPoints = new int[,] {
+      { 395, 86  }, { 477, 107 }, { 545, 157 }, { 591, 229 }, { 623, 319 }, // 1
+      { 627, 404 }, { 599, 484 }, { 546, 551 }, { 471, 606 }, { 390, 637 },
+      { 304, 637 }, { 226, 605 }, { 149, 551 }, { 95,  485 }, { 70,  403 },
+      { 74,  319 }, { 103, 229 }, { 149, 157 }, { 218, 107 }, { 299, 87  },
+      { 348, 139 }, { 425, 149 }, { 487, 165 }, { 524, 219 }, { 555, 290 }, // 2
+      { 572, 366 }, { 575, 431 }, { 535, 482 }, { 477, 534 }, { 409, 574 },
+      { 348, 595 }, { 286, 573 }, { 220, 536 }, { 163, 483 }, { 123, 431 },
+      { 125, 366 }, { 141, 292 }, { 172, 220 }, { 209, 166 }, { 270, 148 },
+      { 386, 199 }, { 456, 210 }, { 487, 272 }, { 511, 346 }, { 523, 414 }, // 3
+      { 473, 463 }, { 410, 509 }, { 348, 541 }, { 286, 509 }, { 223, 464 },
+      { 173, 415 }, { 185, 347 }, { 209, 273 }, { 240, 209 }, { 310, 199 },
+      { 348, 259 }, { 418, 262 }, { 442, 327 }, { 461, 394 }, { 406, 437 }, // 4
+      { 348, 476 }, { 290, 438 }, { 236, 394 }, { 253, 327 }, { 278, 262 },
+      { 379, 314 }, { 400, 375 }, { 349, 412 }, { 297, 375 }, { 316, 314 }, // 5
+      { 348, 358 }                                                          // 6
+    };
+
+    private static double[,] points;
+    private static void InitPoints() {
+      points = new double[handDrawnPoints.GetLength(0), 4];
+      for (int i = 0; i < handDrawnPoints.GetLength(0); i++) {
+        var x = points[i,0] = (((double)handDrawnPoints[i,0])-70.0)/557.0;
+        var y = points[i, 1] = (((double)handDrawnPoints[i, 1]) - 86) / 551.0;
+
+        x = x * 2 - 1;
+        if (x == 0) x += 0.00001;
+        y = y * 2 - 1;
+
+        var angle = points[i, 2] = Math.Tanh(y / x);
```

Comment:
It looks like you forgot to decide between calculating the `angle`/`dist` variables in `InitPoints` versus calculating them dynamically in `GetProjectedLEDPointParametric`. I'm assuming the two different sets of `x`/`y` coordinates complicated matters, though I'm not opposed to storing both in separate static variables, or calculating one from the other in `GetProjectedLEDPointParametric ` as you're currently doing, if that's easier